tests/test_dcargs.py

```diff
@@ -548,3 +548,37 @@ def main() -> argparse.ArgumentParser:
         return parser
 
     assert isinstance(tyro.cli(main, args=[]), argparse.ArgumentParser)
+
+
+def test_return_unknown_args() -> None:
+    @dataclasses.dataclass
+    class A:
+        x: int = 0
+
+    a, unknown_args = tyro.cli(
+        A, args=["positional", "--x", "5", "--y", "7"], return_unknown_args=True
+    )
+    assert a == A(x=5)
+    assert unknown_args == ["positional", "--y", "7"]
+
+
+def test_unknown_args_with_arg_fixing() -> None:
+    @dataclasses.dataclass
+    class A:
+        x: int = 0
+
+    a, unknown_args = tyro.cli(
+        A,
+        args=["--x", "5", "--a_b", "--a-c"],
+        return_unknown_args=True,
+    )
+    assert a == A(x=5)
+    # Should return the unfixed arguments
+    assert unknown_args == ["--a_b", "--a-c"]
+
+    # If there's an argument that's unambigous then we should raise an error.
```

Comment:
typo: I think `unambigous` here should be `ambiguous`

Reply:
Ha! Good catch 😅